### PR TITLE
Define top level Permission for cli.yml workflow

### PR DESCRIPTION
**Pull request by Akash-Singh04**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #5048 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/cli.yaml [hunk 1]
```diff
@@ -12,6 +12,9 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+permissions:
+  contents: read  # Required to check out the code
+  id-token: write # Required for  authentication methods
```
Comment (zhzhuang-zju) [inline .github/workflows/cli.yaml:17]:
```suggestion
```
permission `id-token: write` is unnecessary

File: .github/workflows/cli.yaml [hunk 2]
```diff
@@ -54,5 +57,4 @@ jobs:
         with:
           cluster-resources: nodes,namespaces,
           namespace-resources: configmaps,pods,svc
-          artifact-name: logs-${{ matrix.k8s}}
-
+          artifact-name: logs-${{ matrix.k8s}}
```
Comment (zhzhuang-zju) [inline .github/workflows/cli.yaml:60]: There is no need to modify it here

**Comment by Akash-Singh04** [comment]
@zhzhuang-zju Your requested changes have been made

File: .github/workflows/cli.yaml [hunk 3]
```diff
@@ -55,4 +57,3 @@ jobs:
           cluster-resources: nodes,namespaces,
           namespace-resources: configmaps,pods,svc
           artifact-name: logs-${{ matrix.k8s}}
-
```
Comment (liangyuanpeng) [inline .github/workflows/cli.yaml:58]: this should not change.

**Review by liangyuanpeng — COMMENTED** [review]
please squash the commits. and then LGTM

File: .github/workflows/cli.yaml [hunk 4]
```diff
@@ -12,6 +12,8 @@ on:
 concurrency:
   group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
+permissions:
+  contents: read  # Required to check out the code
```
Comment (zhzhuang-zju) [inline .github/workflows/cli.yaml:16]:
```suggestion
  contents: read 
```

**Comment by zhzhuang-zju** [comment]
> please squash the commits. and then LGTM

@Akash-Singh04 others LGTM

**Comment by Akash-Singh04** [comment]
@zhzhuang-zju Done

File: .github/workflows/cli.yaml [hunk 5]
```diff
@@ -55,4 +57,4 @@ jobs:
           cluster-resources: nodes,namespaces,
           namespace-resources: configmaps,pods,svc
           artifact-name: logs-${{ matrix.k8s}}
-
+
```
Comment (zhzhuang-zju) [inline .github/workflows/cli.yaml:60]:
```suggestion

```
@Akash-Singh04 
Refer to https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b, a github text file should end with a new line. The current `cli.yaml` file is actually end with two new lines, so you can choose to delete just one of the lines, or leave them alone. What you've done is actually remove a new line and add a line that contains invisible characters

File: .github/workflows/cli.yaml (hunk 3 above)
Comment (RainbowMango) [reply .github/workflows/cli.yaml]: fixed.

File: .github/workflows/cli.yaml (hunk 5 above)
Comment (RainbowMango) [reply .github/workflows/cli.yaml]: done.

**Comment by RainbowMango** [comment]
Just helped rebase and resolved the left comments. Please take another look.

**Comment by liangyuanpeng** [comment]
https://github.com/karmada-io/karmada/pull/5079/files#r1665574174 is not resolved 

/appvoe
leave lgtm to @zhzhuang-zju

**Comment by liangyuanpeng** [comment]
/approve

thanks you

**Pull request merged**